### PR TITLE
Add section for interactive API

### PR DESCRIPTION
**Pull request by aoudiamoncef**

### CODE REVIEW

**Comment by aoudiamoncef** [comment]
Preview https://massa--110.org.readthedocs.build/en/110/technical-doc/api.html

**Comment by qdrn** [comment]
Although not directly related I think this sentence is more confusing than helpful `This crate exposes Rust methods (through the `Endpoints` trait) as JSON-RPC API methods (thanks to the `ParityJSON-RPC
<https://github.com/paritytech/jsonrpc>`_ crate).`

**Comment by qdrn** [comment]
```
.. warning::
    Massa has a new experimental API with both Http and WebSocket support: Default port: 33036
    - **Http**: used for node management and blockchain interactions. e.g. http://localhost:33036
    - **WebSocket**: used for streaming blockchain events. e.g. ws://localhost:33036
```

I'd put this warning at the end of the paragraph.

File: technical-doc/api.rst [hunk 1]
```diff
@@ -23,13 +23,15 @@ Massa JSON-RPC API is splitted in two parts :
 Find the complete Massa `OpenRPC <https://spec.open-rpc.org/>`_ specification `here
 <https://raw.githubusercontent.com/massalabs/massa/main/massa-node/base_config/openrpc.json>`_.
 
+.. note::
```
Comment (qdrn) [inline technical-doc/api.rst:26]: I'd put a bit more more context. Maybe something like this: `We provide an <link: Interactive API specification> which documents every endpoints available and allows you to play with the API`.

**Review by qdrn — CHANGES_REQUESTED** [review]

**Comment by aoudiamoncef** [comment]
> ```
> .. warning::
>     Massa has a new experimental API with both Http and WebSocket support: Default port: 33036
>     - **Http**: used for node management and blockchain interactions. e.g. http://localhost:33036
>     - **WebSocket**: used for streaming blockchain events. e.g. ws://localhost:33036
> ```
> 
> I'd put this warning at the end of the paragraph.

Agree, I hesitated to put it as you say.

File: technical-doc/api.rst (hunk 1 above)
Comment (aoudiamoncef) [reply technical-doc/api.rst:26]: Yep

**Comment by qdrn** [comment]
What's going on with the localised files ?

**Comment by aoudiamoncef** [comment]
> What's going on with the localised files ?
I fixed It. quite strange behaviour

**Comment by qdrn** [comment]
It would be nice to finish this PR as the documentation regarding the API is quite unclear.

**Comment by qdrn** [comment]
It would be nice to mention the public RPC that we provide as well in this documentation (test.massa.net/api/v2).

**Comment by aoudiamoncef** [comment]
> It would be also nice to mention the public RPC that we provide as well in this documentation (test.massa.net/api/v2).

I did a complete explanation about the usage of personal vs Massa node

**Comment by aoudiamoncef** [comment]
Hi @qdrn , this PR should prioritized as it solves lot of users issues.

**Comment by qdrn** [comment]
> Hi @qdrn , this PR should prioritised as It solves lot of users issues.

I agree. Most of my comments are still unresolved.

**Review by qdrn — CHANGES_REQUESTED** [review]

File: technical-doc/api.rst [hunk 2]
```diff
@@ -5,114 +5,155 @@
...
-  =Enter%20Massa%20JSON-RPC%20server%20URL&uiSchema[appBar][ui:logoUrl]=
-  https://massa.net/favicons/favicon.ico&uiSchema[appBar][ui:splitView]=
-  false&uiSchema[appBar][ui:darkMode]=false&uiSchema[appBar][ui:title]=
-  Massa&uiSchema[appBar][ui:examplesDropdown]=false&uiSchema[methods]
-  [ui:defaultExpanded]=false&uiSchema[methods][ui:methodPlugins]=true&uiSchema
-  [params][ui:defaultExpanded]=false>`_.
-- **Postman**: use `Postman collection <https://github.com/massalabs/massa-docs/tree/main/postman>`_.
+**JavaScript**: see `massa-web3.js <https://github.com/massalabs/massa-web3>`_.
+
+**Postman**: is a tool for software developers to create, test and debug API requests. You can create collections of requests in Postman workspaces for easy collaboration on projects.
```
Comment (qdrn) [inline technical-doc/api.rst:34]:
```suggestion
**Postman**: is a tool for software developers to create, test and debug API requests.
```
I would remove this sentence. It does not add much.

File: technical-doc/api.rst [hunk 3]
```diff
@@ -5,114 +5,155 @@
...
-  Massa&uiSchema[appBar][ui:examplesDropdown]=false&uiSchema[methods]
-  [ui:defaultExpanded]=false&uiSchema[methods][ui:methodPlugins]=true&uiSchema
-  [params][ui:defaultExpanded]=false>`_.
-- **Postman**: use `Postman collection <https://github.com/massalabs/massa-docs/tree/main/postman>`_.
+**JavaScript**: see `massa-web3.js <https://github.com/massalabs/massa-web3>`_.
+
+**Postman**: is a tool for software developers to create, test and debug API requests. You can create collections of requests in Postman workspaces for easy collaboration on projects. 
+For more information, please visit `Postman learning center <https://learning.postman.com/docs/getting-started/introduction/>`_.
+
+Find all maintained Massa Postman collections in our official `workspace <https://www.postman.com/massalabs>`_.
```
Comment (qdrn) [inline technical-doc/api.rst:37]: It's weird to skip a line before as it's then rendered on another line which breaks the consistency with the javascript bullet point.

File: technical-doc/api.rst [hunk 4]
```diff
@@ -879,307 +1442,1820 @@ associated to a given hash(s). - Query:
...
+                        "content_creator_address": "AU1WqP6GTsR8w4eyRPHQkrQhujmRb3hmLaQt3HZfeTVHWVZade48",
+                        "id": "B122ByHzPVJ3QFwmuYcZ4vZYzq6rfkqx7BJSJdFNHWp9j2o5Fpxv"
                     },
                     "operations": [
-                        "177bzpUmukLarBiRGcTCDE63xqc5nkAKUja414HDmsNS2T3Gy",
-                        "1G9oj5pX7Ruj1w6cT9DXY2KMNKM9JgHVfA6JN5nhZchnHCm6s",
-                        ...
-                        "2wX8fv5WaPpEXFYPbzmyeShVMRNdcqGCxnUCdfXKW2LrVvpkM1",
-                        "2wdwXBXxpBeQuPtT5kkEiV3AKsTqWkzN9QLHXskDsZgD8XpdTN"
+                        "O116jye7zBAttWHfCufcMVQEQaa9JMTqv6To5mE2CzmzBs7cZ7W",
```
Comment (qdrn) [inline technical-doc/api.rst:1737]: There's way too much information. This makes the page gigantic and unusable.

File: technical-doc/api.rst [hunk 5]
```diff
@@ -610,94 +645,622 @@ Get information about `address <https://docs.massa.net/en/latest/general-doc/arc
...
+            "address": "AU12gAkmGeozFceJD4tQmbVvihYdX2KyWZcYLL8xdYZeP4EuWYdex",
+            "thread": 27,
+            "final_balance": "153.519945908",
+            "final_roll_count": 145,
+            "final_datastore_keys": [],
+            "candidate_balance": "153.755240036",
+            "candidate_roll_count": 145,
+            "candidate_datastore_keys": [],
+            "deferred_credits": [],
+            "next_block_draws": [
```
Comment (qdrn) [inline technical-doc/api.rst:668]: Same remark as for operations.

File: technical-doc/api.rst [hunk 6]
```diff
@@ -5,114 +5,155 @@
...
+
+**Postman**: is a tool for software developers to create, test and debug API requests. You can create collections of requests in Postman workspaces for easy collaboration on projects. 
+For more information, please visit `Postman learning center <https://learning.postman.com/docs/getting-started/introduction/>`_.
+
+Find all maintained Massa Postman collections in our official `workspace <https://www.postman.com/massalabs>`_.
+
+
+Using Massa Offical Node vs Running Your Own Node on Testnet
+------------------------------------------------------------
+If you want to deploy Smart Contracts on the Massa Testnet, you have two options: using the Massa official node or running your own node.
```
Comment (qdrn) [inline technical-doc/api.rst:42]: This is not specific to smart contracts. Also you don't need to use capitalised letters.

File: technical-doc/api.rst [hunk 7]
```diff
@@ -5,114 +5,155 @@
...
+
+**Option 1: Using Massa Official Node**
+
+The Massa official node is readily available and easy to use. In order to deploy your SmartContracts, you just need to access https://test.massa.net/api/v2 from your application code, which will allow you to interact with the blockchain network.
+
+However, keep in mind that when you use the Massa official node, your application is relying on a centralized entity, which can pose issues for decentralization efforts. Additionally, a large number of applications may be utilizing the same node, leading to lower quality of service.
```
Comment (qdrn) [inline technical-doc/api.rst:48]: Let's keep it focused.

File: technical-doc/api.rst [hunk 8]
```diff
@@ -5,114 +5,155 @@
...
-- **Postman**: use `Postman collection <https://github.com/massalabs/massa-docs/tree/main/postman>`_.
+**JavaScript**: see `massa-web3.js <https://github.com/massalabs/massa-web3>`_.
+
+**Postman**: is a tool for software developers to create, test and debug API requests. You can create collections of requests in Postman workspaces for easy collaboration on projects. 
+For more information, please visit `Postman learning center <https://learning.postman.com/docs/getting-started/introduction/>`_.
+
+Find all maintained Massa Postman collections in our official `workspace <https://www.postman.com/massalabs>`_.
+
+
+Using Massa Offical Node vs Running Your Own Node on Testnet
```
Comment (qdrn) [inline technical-doc/api.rst:40]: I would simply provide a table like what is done here: https://chainlist.org/chain/1
We can also add some community RPCs as they are developped.

**Review by qdrn — CHANGES_REQUESTED** [review]

File: technical-doc/api.rst (hunk 8 above)
Comment (aoudiamoncef) [reply technical-doc/api.rst:40]: This is a good idea, I talked about it with tech ambassadors.

Comment (aoudiamoncef) [reply technical-doc/api.rst:40]: image

features: JsonRPC, gRPC,WebSocket

File: technical-doc/api.rst (hunk 5 above)
Comment (aoudiamoncef) [reply technical-doc/api.rst:496]: done

File: technical-doc/api.rst (hunk 6 above)
Comment (aoudiamoncef) [reply technical-doc/api.rst]: done

File: technical-doc/api.rst (hunk 7 above)
Comment (aoudiamoncef) [reply technical-doc/api.rst]: done

File: technical-doc/api.rst (hunk 4 above)
Comment (aoudiamoncef) [reply technical-doc/api.rst:705]: done

**Comment by aoudiamoncef** [comment]
https://massa--110.org.readthedocs.build/en/110/technical-doc/api.html

File: technical-doc/api.rst [hunk 9]
```diff
@@ -5,114 +5,155 @@
...
 
 - **Private API**: used for node management. Default port: 33034 e.g. http://localhost:33034
 - **Public API**: used for blockchain interactions. Default port: 33035 e.g. http://localhost:33035
 
+Find the complete Massa `OpenRPC <https://spec.open-rpc.org/>`_ specification `here
+<https://raw.githubusercontent.com/massalabs/massa/main/massa-node/base_config/openrpc.json>`_.
+
+.. note::
+
+   We provide an `Interactive API Specification <https://playground.open-rpc.org/?schemaUrl=https://test.massa.net/api/v2&uiSchema[appBar][ui:input]=false&uiSchema[appBar][ui:inputPlaceholder]=Enter%20Massa%20JSON-RPC%20server%20URL&uiSchema[appBar][ui:logoUrl]=https://massa.net/favicons/favicon.ico&uiSchema[appBar][ui:splitView]=false&uiSchema[appBar][ui:darkMode]=false&uiSchema[appBar][ui:title]=Massa&uiSchema[appBar][ui:examplesDropdown]=false&uiSchema[methods][ui:defaultExpanded]=false&uiSchema[methods][ui:methodPlugins]=true&uiSchema[params][ui:defaultExpanded]=false>`_ which documents every available method and allows you to play with the API.
```
[the remark this answers is not in the document]
Comment (qdrn) [reply technical-doc/api.rst:20]: You still need to resolve this one @aoudiamoncef !

**Review by qdrn — CHANGES_REQUESTED** [review]
There are still a lot of comments that are not resolved!

File: technical-doc/api.rst (hunk 7 above)
Comment (aoudiamoncef) [reply technical-doc/api.rst]: For this part, I removed the SC and made it more general

File: technical-doc/api.rst (hunk 8 above)
Comment (aoudiamoncef) [reply technical-doc/api.rst]: I kept the Postman Workspace which provided a real time documentation of all our stable and experimental features.

I added a table with Massa official node, and launched a conversation with tech ambassadors about adding community nodes to this section later

File: technical-doc/api.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (aoudiamoncef) [reply technical-doc/api.rst:20]: I don't see anything anormal here: [massa--110.org.readthedocs.build/en/110/technical-doc/api.html](https://massa--110.org.readthedocs.build/en/110/technical-doc/api.html)

> There are still a lot of comments that are not resolved!

I didn't know that the linter was updated ? it always fail no ?

**Comment by aoudiamoncef** [comment]
Hi @qdrn 

should I remove this file `/home/runner/work/massa-docs/massa-docs/web3-dev/smart-contracts/smart-contract-example-sum.rst: WARNING: document isn't included in any toctree` which make the CI fails ? 

It was a hidden behaviours... for me the CI is always failing, but I checked that you rely on the first stage of the CI(lint) to say that I added smells. 

We could easily remove this ambiguity by removing the warning which makes the make command fail.

File: technical-doc/api.rst (hunk 7 above)
Comment (aoudiamoncef) [reply technical-doc/api.rst]: I switched to small sentences.

**Comment by qdrn** [comment]
> should I remove this file `/home/runner/work/massa-docs/massa-docs/web3-dev/smart-contracts/smart-contract-example-sum.rst: WARNING: document isn't included in any toctree` which makes the CI fails ?

No. It's being handled in another issue. I just want to avoid adding linter errors!

**Review by qdrn — APPROVED** [review]

**Comment by qdrn** [comment]
I've removed the section and approved. Let's merge it if it passes CI.

**Pull request merged**